model/src/w3iogomd.F90

```diff
@@ -1562,8 +1616,39 @@ SUBROUTINE W3OUTG ( A, FLPART, FLOUTG, FLOUTG2 )
             USSCO=FKD*SIG(IK)*WN(IK,ISEA)*COSH(2.*KD)
             BHD(JSEA) = BHD(JSEA) +                             &
                 GRAV*WN(IK,ISEA) * EBD(IK,JSEA) / (SINH(2.*KD))
+#ifdef CESMCOUPLED
```

Comment:
Agreed; this was meant to minimize any impact w/ the current code. Bringing in features like this for wider use would definitely be something we can work on.

Reply:
The calculations require the surface layer depth HSL, set as 0.2* HML, where HML is the mixed layer depth imported from the Ocean. We don't currently export this field from the Ocean in UWM so this was kept behind the cesm ifdef.

Reply:
I'll have to look more but it seems like the "surface stokes drift" doesn't depend on HSL and we actually already have a surface stokes drift calculation (USSX,USSY) so understanding how those are different will be important at some point.  We might should add HML as an import field in the standalone mode to WW3 then?

Reply:
@JessicaMeixner-NOAA - again I think this is a question for CC and Lettie. I'm not sure what their github ids are. @alperaltuntas - are you working on this particular part of the code base?

Reply:
These are the stokes drift calculations added by @qingli411, which are used in CESM's legacy OCN-WAV coupling that is based on passing the computed stokes drifts and langmuir enhancement factor to CVMix (through POP2 or MOM6). These changes were initially introduced to NCAR's old WW3 fork based on v.3.14.  @qingli411 can elaborate more.

Reply:
These Stokes drift calculations were added to enable the Langmuir turbulence parameterization in CESM through CVMix ([Li et al., 2016](https://doi.org/10.1016/j.ocemod.2015.07.020); [Li and Fox-Kemper 2007](https://doi.org/10.1175/JPO-D-17-0085.1)), which requires some scalar quantities such as the surface layer averaged Langmuir number and the Langmuir multiplier. These quantities can be diagnosed from the full profile of the Stokes drift. But they were added here in WW3 because (1) the full profile of Stokes drift is an array with depth rather than a scalar which is more expensive to pass from WW3 to the ocean model, and (2) the vertical resolution of the ocean model is too coarse to resolve the shape of the Stokes drift profile. 

@JessicaMeixner-NOAA The surface Stokes drift doesn't depend on the surface layer depth HSL (defined as 1/5 the mixed layer depth HML). It is calculated in newer versions of WW3 but was not there in the old version we had in CESM based on v3.14. Actually I think I ported this specific feature from v4.18. A difference in the calculation is that we added the contribution from a f^{-5} spectral tail, which seems to be commented out in newer versions of WW3. The surface layer averaged Stokes drift depends on HSL, but I'm not sure if it makes sense to add it as a standard WW3 output -- as you mentioned we will need to add HML as an import field for that but HML is not used in the calculation of any prognostic quantities in WW3. I think right now the use of these variables and the way they were implemented are really specific to CESM (or CVMix to be more accurate). Ideally, it would be nice to have a standard and easy way to pass the full Stokes drift profile to the ocean. Then these quantities can be diagnosed in the ocean model.

Reply:
@qingli411 if this does not make sense as an output of WW3 perhaps it should be moved to just a calculation of an import/export in the nuopc cap instead of being in the output computation routine?

Reply:
@JessicaMeixner-NOAA I think it does make sense to have these variables as output for CESM -- when running CESM coupled with WW3, it would be good to have some of them in the output. But I'm just not sure if it makes sense to add them to the standard output of WW3 if other users do not use them... So, perhaps it is OK to keep them behind the `CESMCOUPLED` ifdefs for now? In the future, I think it would be good to have a more general way to pass the full Stokes drift profile information to the ocean, e.g., something like what @breichl did? Then these quantities can be diagnosed from the Stokes drift profile in the ocean model.

Reply:
That being said, if this is temporary and we make an issue to remove this as you move to use the full Stokes profile, then perhaps just making an issue is fine.  However, then we have the issue that these are taking up places in the output array structures and so the next person who adds a variable is going to add something after that and then that gets messy to reconcile, so maybe this should use different indices so it'd be easier for others to add new variables in the meantime?

Reply:
@JessicaMeixner-NOAA OK I see. How about this? We can add a new standard output in WW3 for a depth averaged Stokes drift which requires a new import field of a depth. We don't need to specify the meaning of this depth, i.e., it can be 1/5 of the mixed layer depth as we've been using in CESM, but it can also be any depth of interest for other users. In the limit that this depth approaches zero, the depth averaged Stokes drift should be equivalent to the surface Stokes drift. In the limit that this depth approaches the water depth, the depth averaged Stokes drift should be the total Stokes transport divided by the water depth. I think both surface Stokes drift and total Stokes transport are standard output of WW3. This depth averaged Stokes drift can serve as another variable to describe the Stokes drift, which is potentially useful to other users. Then, I think the calculation of all other variables such as the Langmuir multiplier can be moved to the nuopc cap. What do you think?

There is another issue that whether we should include the contribution of a spectral tail to the Stokes drift. We are including the contribution from a f^{-5} spectral tail when using it in CESM. But I suspect that the Langmuir multiplier might not be very sensitive to this choice, so the effect of dropping this contribution is probably small...

Reply:
@qingli411 that sounds good to me.  

In terms of the tail to the Stokes drift, I'd recommend making an issue and then creating a flag to add it or not in a separate PR. My guess is others will be just as interested as you to have the option to add the tail or not.  But this way we only have one calculation of surface Stokes drift instead of multiple.

Reply:
@JessicaMeixner-NOAA Sounds good!